etsin_finder/resources/download_resources.py

```diff
@@ -120,11 +123,21 @@ def post(self):
 
         """
         self.parser.add_argument('scope', type=str, action='append', required=False)
+        self.parser.add_argument('identifier', type=str, required=False)
         args = self.parser.parse_args(strict=True)
         cr_id = args.get('cr_id')
         check_download_permission(cr_id)
 
         scope = args.get('scope')
+        identifier = args.get('identifier')
+        common_service = MetaxCommonAPIService()
+        resp, status = common_service.get_single_directory(identifier, {})
+
+        if status == 200 and resp is not None:
```

Comment:
Return an error if `get_single_directory` status is not 200. Currently the code calls `download_service.post_request` if `common_service.get_single_directory` fails.